routes/agreement-1/agreement-1.controller.js

```diff
@@ -21,40 +22,66 @@ function getRandomString() {
 
 const flagHtml = `
     <img src="public/img/GOC_colour_en.png" alt="Symbol of the Government of Canada" width="300px">
-`;
+`
 
 const wordmarkHtml = `
     <img src="public/img/Canwordmark_colour.png" alt="Government of Canada" width="150px">
-`;
+`
 
 module.exports = app => {
   const name = 'agreement-1'
   const route = routeUtils.getRouteByName(name)
 
   routeUtils.addViewPath(app, path.join(__dirname, './'))
 
-  app
-    .get(route.path, (req, res) => {
-      var nextRoute = getNextRoute(name).path;
-      var randomString = getRandomString()
-      var docxFilename = 'agreement-' + randomString + '.docx'
-      res.render(
-        name + `-${i18n.getLocale(req)}`,
-        { 
-          ...routeUtils.getViewData(req, {}), 
-          nextRoute: nextRoute,
-          docxFilename: docxFilename,
-        }, 
-        function(err, html) {
-          if(err) {
-            console.log(err)
-          }
-        const startIndex = html.indexOf("<h1>");
-        const endIndex = html.indexOf("</main>");
-        const htmlDoc = flagHtml + html.slice(startIndex, endIndex) + wordmarkHtml;
-        nodePandoc(htmlDoc, "-f html -t docx -o public/documents/" + docxFilename, callback)
-        res.send(html);
-      })
+  app.get(route.path, (req, res) => {
+    var nextRoute = getNextRoute(name).path
+    var randomString = getRandomString()
+    var docxFilename = 'agreement-' + randomString + '.docx'
 
+    let link = url.format({
+      protocol: req.protocol,
+      host: req.get('Host'),
+      pathname: routeUtils.getRouteByName('questions-1').path,
     })
+
+    const data = routeUtils.getViewData(req, {}).data
+
+    link =
+      link +
+      '?' +
+      Object.keys(data)
+        .filter(key => key !== '_csrf' && data[`${key}`] !== '')
+        .map(
+          key =>
+            `${encodeURIComponent(key)}=${encodeURIComponent(data[`${key}`])}`,
```

Comment:
Node has a standard library to do this: https://nodejs.org/api/querystring.html

Reply:
actually it turns out that I can just pass an object with the params to `url.format()` and it will parse them 👍 
```
const link = url.format({
      protocol: req.protocol,
      host: req.get('Host'),
      pathname: routeUtils.getRouteByName('questions-1').path,
      query: queryParams,
    })
```